### DISCUSSION TITLE
Configuring `next lint` to run using `--quiet` at build time

### DISCUSSION THREAD

**Discussion by nbouvrette** [vercel/next.js#45853] [Ideas]
### Describe the feature you'd like to request

It looks like the new `--quiet` feature for `next lint` was added a while back here:

https://github.com/vercel/next.js/pull/26280/files

Unfortunately, I could not find any option where you could configure this in your Next.js configuration file so that this setting becomes the default `next lint` behavior, especially when running `npm run build` where you cannot provide the `--quiet` flag.

A common use case would be the false warnings of `react-hooks/exhaustive-deps` which can become quite annoying, but still can be useful when used in the IDE.

### Describe the solution you'd like

I was thinking of adding a simple config like this in the Next.js config (next.config.js):

```
  eslint: {
    // Warnings are not useful, so we leave them only in the IDE.
    reportErrorsOnly: true,
  },
```

### Describe alternatives you've considered

Can't think of any since `next lint` is run part of the build.. other than disabling it completely which comes with bigger drawbacks than just living with the false positive logs.

**Comment by icyJoseph** [comment]
Run it with `npx next lint --quiet`:

```console
npx next lint
warn  - You have enabled experimental feature (appDir) in next.config.js.
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.
info  - Thank you for testing `appDir` please leave your feedback at https://nextjs.link/app-feedback


./src/app/page.tsx
11:7  Warning: The 'z' object makes the dependencies of useEffect Hook (at line 14) change on every render. Move it inside the useEffect callback. Alternatively, wrap the initialization of 'z' in its own useMemo() Hook.  react-hooks/exhaustive-deps

info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
```

And with `--quiet`:

```console
npx next lint --quiet
warn  - You have enabled experimental feature (appDir) in next.config.js.
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.
info  - Thank you for testing `appDir` please leave your feedback at https://nextjs.link/app-feedback

✔ No ESLint warnings or errors
```

And with an actual ESLINT error, not just a warning:

```console
npx next lint --quiet
warn  - You have enabled experimental feature (appDir) in next.config.js.
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.
info  - Thank you for testing `appDir` please leave your feedback at https://nextjs.link/app-feedback


./src/app/page.tsx
14:5  Error: React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render.  react-hooks/rules-of-hooks
```

## Edit

Ah never mind you want to `--quiet` it as part of the build pipeline... misread that part, my bad! 🤦 🤦  Leaving the above just in case anyone wants that.

## Edit 2

To accomplish OPs request, one could stretch and do:

```
npx next lint --quiet && npx next build --no-lint
```

Brittle, but it works, for now.

**Reply by nbouvrette** [reply]
I know, this is not the goal of this issue - it's to be able to do this automatically when running `npm run build`

**Reply by nbouvrette** [reply]
I added clarifications in the discussion.

**Reply by icyJoseph** [reply]
Yeah, I noticed it right after I posted n_n

**Reply by icyJoseph** [reply]
I couldn't find any way to get around this, in code, but to do perhaps run something like:

```
npx next lint --quiet && npx next build --no-lint
```

If there's errors, they'll float out of the first command, and block the second command, not ideal, but I guess it works.

**Reply by nbouvrette** [reply]
Yeah, that is a decent workaround. I still think someone should implement this feature - I bet it would take a few minutes to implement, but I am not familiar enough with the PR process to do this myself.
